### PR TITLE
Improved Docker Registry Support (fixing)

### PR DESCRIPTION
**Pull request by rsandell**
Adapting PR #7 and #5 to be backwards compatible with old data.

Also taking care of some other PR comments

@reviewbybees esp @michaelneale who reviewed the original PR

### CODE REVIEW

**Comment by rsandell** [comment]
[This diff](https://github.com/jenkinsci/dockerhub-notification-plugin/compare/1b26149...adapt-pull-7) shows only my changes including PR #10

File: src/main/java/org/jenkinsci/plugins/registry/notification/DockerHubTrigger.java [hunk 1]
```diff
@@ -135,4 +150,57 @@ public String getDisplayName() {
...
+                                     TriggerOnSpecifiedImageNames.class);
+            xs.addCompatibilityAlias("org.jenkinsci.plugins.dockerhub.notification.TriggerStore$TriggerEntry",
+                                     TriggerStore.TriggerEntry.class);
+            xs.addCompatibilityAlias("org.jenkinsci.plugins.dockerhub.notification.TriggerStore$TriggerEntry$RunEntry",
+                                     TriggerStore.TriggerEntry.RunEntry.class);
+        }
+    }
+
+    /**
+     * Hack to get around the fact that {@link Fingerprint#XSTREAM} has private access.
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/registry/notification/DockerHubTrigger.java]: Are you filing a core PR to fix that?

**Comment by jglick** [comment]
:bee: on the added commits

**Comment by lguminski** [comment]
@reviewbybees @jglick This pull request does not originate from a CloudBess employee. It originates from HolidayCheck AG in cooperation with ContainerSolutions BV. I am sorry for not being here for a while, but I missed the entire conversations. Thank you for taking this pull request further. Please let me know if you need any assistance.

**Comment by rsandell** [comment]
@lguminski the later commits (from me) originates from a CloudBees employee ;)

File: src/main/java/org/jenkinsci/plugins/registry/notification/DockerHubTrigger.java (hunk 1 above)
Comment (rsandell) [inline src/main/java/org/jenkinsci/plugins/registry/notification/DockerHubTrigger.java]: Yes I was planning to do that.

**Comment by lguminski** [comment]
@rsandell I noticed :) I meant the original pull request. That's great that you are working on it. Do you think you can merge it soon?

**Comment by rsandell** [comment]
@lguminski I hope to have it merged and released very soon, at least before the end of the week. I noticed I need to add a `CrumbExclusion` in a separate pr as well before release, somehow that was lost in the very first release of the plugin so it doesn't work well under a secured Jenkins.

**Comment by lguminski** [comment]
@rsandell I understand. Thanks.

File: src/main/java/org/jenkinsci/plugins/registry/notification/webhook/JSONWebHook.java [hunk 2]
```diff
@@ -21,55 +21,33 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
-package org.jenkinsci.plugins.dockerhub.notification;
+package org.jenkinsci.plugins.registry.notification.webhook;
```
Comment (michaelneale) [inline src/main/java/org/jenkinsci/plugins/registry/notification/webhook/JSONWebHook.java]: No docker in this namespace seems a bit odd given it looks like it is docker specific?

File: src/main/java/org/jenkinsci/plugins/registry/notification/webhook/WebHookPayload.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.jenkinsci.plugins.registry.notification.webhook;
```
Comment (michaelneale) [inline src/main/java/org/jenkinsci/plugins/registry/notification/webhook/WebHookPayload.java]: Is this intended to be generic beyond the docker hub registry or webhook?

**Comment by michaelneale** [comment]
:bee: this is great. I thought some of the generic sounding package name refactoring seemed a bit odd, but I am sure there is a reason.

File: src/main/java/org/jenkinsci/plugins/registry/notification/webhook/WebHookPayload.java (hunk 3 above)
Comment (rsandell) [inline src/main/java/org/jenkinsci/plugins/registry/notification/webhook/WebHookPayload.java]: I guess so, I haven't touched the refactoring from the original PR beyond trying to make the data model backwards compatible.

File: src/main/java/org/jenkinsci/plugins/registry/notification/DockerHubTrigger.java (hunk 1 above)
Comment (rsandell) [inline src/main/java/org/jenkinsci/plugins/registry/notification/DockerHubTrigger.java]: https://github.com/jenkinsci/jenkins/pull/2163

**Comment by rsandell** [comment]
@reviewbybees done

**Pull request merged**